tests/end_to_end.rs

```diff
@@ -1316,3 +1329,117 @@ async fn test_delete_statement() {
+    let expected = vec![
+        "+-----------------+----------------+------------------+-----------+------------+",
+        "| some_bool_value | some_int_value | some_other_value | some_time | some_value |",
+        "+-----------------+----------------+------------------+-----------+------------+",
+        "|                 | 5555           |                  |           | 32         |",
+        "|                 | 5555           |                  |           | 31         |",
+        "|                 |                |                  |           | 40         |",
+        "+-----------------+----------------+------------------+-----------+------------+",
+    ];
+    assert_batches_eq!(expected, &results);
+
+    //
+    // Execute UPDATE without a selection, creating table_version 7 with all new partitions
+    //
+    let plan = context
+        .plan_query("UPDATE test_table SET some_value = 42")
+        .await
+        .unwrap();
+    context.collect(plan).await.unwrap();
+
+    assert_partition_ids(&context, 7, vec![7]).await;
+
+    // Verify results
+    let plan = context
+        .plan_query("SELECT * FROM test_table")
+        .await
+        .unwrap();
+    let results = context.collect(plan).await.unwrap();
+
+    let expected = vec![
+        "+-----------------+----------------+------------------+---------------------+------------+",
+        "| some_bool_value | some_int_value | some_other_value | some_time           | some_value |",
+        "+-----------------+----------------+------------------+---------------------+------------+",
+        "|                 |                |                  |                     | 42         |",
+        "|                 |                |                  |                     | 42         |",
+        "|                 |                |                  |                     | 42         |",
+        "|                 |                |                  |                     | 42         |",
+        "|                 |                |                  |                     | 42         |",
+        "|                 |                |                  |                     | 42         |",
+        "|                 | 5555           |                  | 2022-01-01 20:01:01 | 42         |",
+        "|                 | 5555           |                  | 2022-01-01 20:02:02 | 42         |",
+        "|                 | 3333           |                  | 2022-01-01 20:03:03 | 42         |",
+        "|                 | 5555           |                  |                     | 42         |",
+        "|                 | 5555           |                  |                     | 42         |",
+        "|                 |                |                  |                     | 42         |",
+        "+-----------------+----------------+------------------+---------------------+------------+",
+    ];
+    assert_batches_eq!(expected, &results);
```

Comment:
These tests look good. Worth testing some extra error/corner cases here:

  - requires an implicit cast (e.g. `SET some_time = '2022-01-01 21:21:21')
  - results in a type mismatch (e.g. `SET some_int_value = 'nope'`)
  - references a nonexistent column (`SET nonexistent = 42`)
  - doesn't change anything (`SET some_bool_value = TRUE WHERE some_value = 200`)
  - redundant update (`SET some_bool_value = TRUE, some_bool_value = (some_int_value = 5555)`)
  - a more complex expression (`SET some_other_value = CASE WHEN some_int_value = 5555 THEN 'five' WHEN some_int_value = 3333 THEN 'three' ELSE 'none' END`)

Reply:
Good idea, added all these variations now.

I've commented out the type mismatch tests for now because, even though they themselves pass, the last case errors out during planning with `Internal("Internal SQL error: \"error returned from database: syntax error at end of input\"")`. Looks like there's something not being closed/cleaned from our side while planing the type mismatch queries.

Reply:
Turns out the issue was that when planning UPDATE we first create a new table version, and only then try to execute the plan into partitions, which fails due to the type mismatch, but leaves the new empty table version hanging.

I've fixed it by creating the new table version only when needed (when linking to newly created partitions).